pkg/k8s/svcdetector/repository.go

```diff
@@ -145,6 +146,33 @@ func (repository defaultServiceRepository) dispatchEvents(eventType int, informe
 // NewDefaultServiceRepository creates ServiceRepository instances
 func NewDefaultServiceRepository(client dynamic.Interface) ServiceRepository {
 	return &defaultServiceRepository{
-		client: client,
+		client:                         client,
+		serviceInformerRestartInterval: time.Minute * 5,
 	}
 }
+
+func runInformerInBg(f func() chan struct{}, timeout time.Duration) {
+	go func() {
+		for {
+			timeoutChan := make(chan struct{})
```

Comment:
why `timeoutChan` has to be struct{} ? It looks that basically any simple type like `bool` or `int` would be ok as well.

Reply:
I also always used bool chans, but here i used whatever  is used in :92 - kubernets lib requires such channel for termination of informer, so naturally, for a very similar purpose I decided to use the same channel type